File: MyMindNotes/app/src/main/java/com/android/mymindnotes/retrofit/ChangeNicknameApi.java
package com.android.mymindnotes.retrofit;

import com.android.mymindnotes.model.ChangeUserNickname;
import com.android.mymindnotes.model.ChangeUserPassword;

import java.util.Map;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.PUT;

public interface ChangeNicknameApi {
    @PUT("/api/member/update/nickname")
    Call<Map<String, Object>> updateUserNickname(@Body ChangeUserNickname changeUserNickname);
}

File: MyMindNotes/data/src/main/java/com/android/mymindnotes/data/retrofit/api/user/ChangeNicknameApi.kt
package com.android.mymindnotes.data.retrofit.api.user

import retrofit2.http.PUT
import com.android.mymindnotes.core.dto.ChangeUserNickname
import com.android.mymindnotes.core.dto.ChangeNicknameResponse
import retrofit2.http.Body

interface ChangeNicknameApi {
    @PUT("/api/member/update/nickname")
    suspend fun updateUserNickname(
        @Body changeUserNickname: ChangeUserNickname
    ): ChangeNicknameResponse
}